homework/index.js

```diff
@@ -30,14 +30,77 @@
+    const tBody = createAndAppend('tBody', table);
+    const trRepository = createAndAppend('tr', tBody, { class: 'table-info' });
+    createAndAppend('td', trRepository, { text: 'Repository: ' });
+    createAndAppend('a', trRepository, {
+      text: repository.name,
+      href: repository.html_url,
+    });
+    const trDescription = createAndAppend('tr', tBody, { class: 'table-info' });
+    createAndAppend('td', trDescription, { text: 'Description: ' });
+    createAndAppend('td', trDescription, { text: repository.description });
+    const trForks = createAndAppend('tr', tBody, { class: 'table-info' });
+    createAndAppend('td', trForks, { text: 'Forks: ' });
+    createAndAppend('td', trForks, { text: repository.forks });
+    const trUpdated = createAndAppend('tr', tBody, { class: 'table-info' });
+    createAndAppend('td', trUpdated, { text: 'Updated: ' });
+    createAndAppend('td', trUpdated, { text: repository.updated_at.toLocaleString() });
+  }
+
+  function renderContributors(rightContainer, repository, root) {
+    rightContainer.innerHTML = '';
+    const contributorsURL = repository.contributors_url;
+    createAndAppend('p', rightContainer, { text: 'Contributions', id: 'contTitle' });
+
+    fetchJSON(contributorsURL, (err, contributors) => {
+      if (err) {
+        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+      }
+
+      contributors.forEach(contributor => {
+        createAndAppend('ul', rightContainer, { class: 'ul' });
+        const li = createAndAppend('li', rightContainer, { class: 'li' });
+        createAndAppend('img', li, { src: contributor.avatar_url, class: 'contImg' });
+        createAndAppend('a', li, {
+          text: contributor.login,
+          href: contributor.html_url,
+          target: '_blank',
+          class: 'contLink',
+        });
+        createAndAppend('div', li, { text: contributor.contributions, class: 'contNumber' });
+      });
+    });
+  }
+
   function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
+    const root = document.getElementById('root');
+
+    fetchJSON(url, (err, repositories) => {
       if (err) {
         createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+        return;
       }
+      const headBox = createAndAppend('div', root, { class: 'header' });
+      createAndAppend('p', headBox, { text: 'HYF Repositories', id: 'headTitle' });
+      const selectBox = createAndAppend('select', headBox, { class: 'selectBox' });
+      const containers = createAndAppend('div', root, { class: 'containers' });
+      const leftContainer = createAndAppend('div', containers, { class: 'leftContainer' });
+      const rightContainer = createAndAppend('div', containers, { class: 'rightContainer' });
+
+      repositories.forEach((repository, index) => {
```

Comment:
The assignment asked for the repositories to be sorted by name.

Reply:
Hello @remarcmij code has been improved based on your comment.
Thanks.